### DISCUSSION TITLE
Event Listener

### DISCUSSION THREAD

**Discussion by wassimkha** [eclipse-theia/theia#10099] [Q&A] [answered]
Hi,

I have a Theia extension and I want to listen to strandard JS event listener, just like the one below:

```js
window.addEventListener("message", (event) => {
        if (event.data && event.data.type == "event_type") {
             // do stuff based on the type, like open files, run commands in terminal, ...
          });
        }
```

I need those events (in the callback function) to be able to open files, run commands, ....

Where would be the best place to put it (i.e were in the extension (browser, common, node folder))? in a separate file?

Any help is greatly appreciated!

**Comment by vince-fugnitto** [comment, marked as answer]
@wassimkha thank you for the discussion, it depends on what you're trying to listen on.

For example, if you need browser (frontend) APIs like `window` you would need to have your listener in `browser` as `node` does not know or should not know about browser APIs.

For additional information, you can take a look at code-organization (which describes what you are allowed to use in which context):
- https://github.com/eclipse-theia/theia/wiki/Code-Organization

**Reply by wassimkha** [reply]
I see, that makes sense. Thank you!
